### ISSUE TITLE
ZNC buffer playback messages are not picked up on

### ISSUE THREAD

**Issue by Yoshi2889** [phergie/phergie-irc-parser#39]
ZNC distributes messages like "Buffer playback started" to the channel using nickname ***. The parser doesn't like * in the nickname and therefore marks the message as invalid:
```
:***!znc@znc.in PRIVMSG #channel:Buffer Playback...
```

Adding * to the list of special characters works, but I'm not sure if this is a valid solution.

**Comment by elazar** [comment]
I think #34 is related.

**Comment by Yoshi2889** [comment]
I was looking at #34 as well, not really sure if related.

I did some googling, and it looks like * might not even be a valid character for a nickname. I'm not sure if ZNC does this intentionally or not, I'll have a chat with them.

**Comment by Yoshi2889** [comment]
```
[21:37:56] <NanoSector> heya, I was working with my bot today which was attached to ZNC, but noticed it wasn't picking up the messages sent out by ***
[21:38:11] <NanoSector> looking at the RFC for the protocol it seems * is not even a valid character for a nickname
[21:38:16] <NanoSector> is this done intentionally?
[21:38:28] <Zarthus> yes, it avoids name colissions
[21:38:42] <Zarthus> you can change the prefix, though
[21:38:50] * Pulec (~pulec@2a01:4f8:110:1463:127::2) joins #znc
[21:38:58] <NanoSector> oh it's fine, I was just wondering if it had a meaning :)
[21:39:12] <NanoSector> thanks :)
[21:39:12] <Zarthus> As long as the IRCD sends it to you it will be a valid nickname, or or less.
[21:39:19] <Zarthus> more or less*
[21:39:35] <Zarthus> there's no real need to doublecheck if the name is rfc compatible
[21:40:09] <NanoSector> yeah, i was using a parser component in my bot which flagged the message as malformed
[21:40:21] <NanoSector> i'll report this to them, thanks :)
```

I'm not sure what the parser component could do about this, since it doesn't know if the message came from ircd or not.

**Comment by svpernova09** [comment]
Resolved with #41
